### ISSUE TITLE
Misc.php returnBytes method not compatible with PHP 7

### ISSUE THREAD

**Issue by GhaziTriki** [ckfinder/ckfinder#302]
We have CKEditor 2.6.0 within ForkCMS running under PHP 7.0. The returnBytes method is not compatible with PHP 7.0. Using the last one in this page makes it all working http://pl.php.net/manual/en/function.ini-get.php. Can this be fixed please?

**Comment by zaak** [comment]
Hi @GhaziTriki. Thanks for reporting. Which version of CKFinder do you use?

**Comment by GhaziTriki** [comment]
@zaak sorry I meant CKFinder v2.6.0

**Comment by JhonnyJason** [comment]
Hi, 
I'm currently dealing with the same propblem.

Has this already been fixed in a newer version?
Or how may I fast fix it? 
I do not really understand what the problem is - just that it is within this file Misc.php within the returnBytes Mehtod - and happens in combination with php7^^

**Comment by JhonnyJason** [comment]
I I found out what GhaziTriki meant - this function appears to be a fix:

```
function returnBytes($val)
{
    assert('1 === preg_match("/^\d+([kmg])?$/i", $val)');
    static $map = array ('k' => 1024, 'm' => 1048576, 'g' => 1073741824);
    return (int)$val * @($map[strtolower(substr($val, -1))] ?? 1);
}
```
Still I do neither understand the Problem nor the Solution xD at least it is in Balance :-)

**Comment by mgsmus** [comment]
Here's the fix for both php5 and php7

```
public static function returnBytes($val) {
    $val = trim($val);
    if (!$val) {
        return 0;
    }
    $last = strtolower($val[strlen($val)-1]);

    // fix
    $val = preg_replace('~\D~', '', $val);

    switch($last) {
        // The 'G' modifier is available since PHP 5.1.0
        case 'g':
            $val *= 1024;
        case 'm':
            $val *= 1024;
        case 'k':
            $val *= 1024;
    }

    return $val;
}
```

**Comment by zaak** [comment]
Fixed. The updated code will be publicly available in the upcoming release.
